subsys/bluetooth/mesh/keys.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/bluetooth/mesh.h>
+#include "keys.h"
+
+int bt_mesh_import_key(enum key_type type, const raw_key_t in, mesh_key_t *out)
+{
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+	psa_set_key_usage_flags(&key_attributes, PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_DECRYPT);
+	psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_VOLATILE);
+
+	switch (type) {
+	case ECB_KEY:
+		psa_set_key_algorithm(&key_attributes, PSA_ALG_ECB_NO_PADDING);
+		break;
+	case CCM_KEY:
+		psa_set_key_algorithm(&key_attributes,
+			PSA_ALG_AEAD_WITH_AT_LEAST_THIS_LENGTH_TAG(PSA_ALG_CCM, 4));
+		break;
+	default:
+		return -EIO;
+	}
+
+	psa_set_key_type(&key_attributes, PSA_KEY_TYPE_AES);
+	psa_set_key_bits(&key_attributes, 128);
+
+	if (psa_import_key(&key_attributes, in, 16, out) != PSA_SUCCESS) {
+		return -EIO;
+	}
+
+	return 0;
+}
```

Comment:
I expect a bit more functions here. My assumption is to keep here import\export (probably something else)functionality for PSA keys. This file should be common for all crypto libraries that provide PSA API. I mean this should work for TF-M lib as well as for mbedTLS.

I'd prefer to avoid inlining this function. This is quite heavy. Inlining such heavy function in all places when keys are imported(when we migrate all keys there will be many places for import) will cause significant flash consumption.

Reply:
seems like I used new types everywhere it had sense. done.